Исчисление высказываний

turing228 opened the pull request:
Closes #5 

- ✔️ Теорема 4.7: доказал утверждение, что P1.. Pn |- alpha;
- ✔️ Теорема 4.8: Правильно доказал базу, разобрав две аксиомы
- ➕ Про упражнения: первая лемма из определения, вторая из предыдущей к ней леммы очевидно следует. Так и было рассказано на лекции

turing228 committed:
first commit

Теорема 4.7: доказал утверждение, что P1.. Pn |- alpha;
Теорема 4.8:
Правильно доказал базу, разобрав две аксиомы

Про упражнения: первая лемма из определения, вторая из предыдущей к ней леммы очевидно следует. Так и было рассказано на лекции

artemohanjanyan commented on lecture3.tex:
```diff
@@ -319,6 +319,35 @@ \section{Теорема о полноте исчисления высказыв
+\begin{proof}Упражнение\end{proof}
+
+\begin{lemma}
+Если $\alpha$ содержит $n$ пропозициональных переменных $P_1 \dots P_n$, то \\
```
Форсированные переносы строк ставить нельзя

artemohanjanyan commented on lecture3.tex:
```diff
@@ -319,6 +319,35 @@ \section{Теорема о полноте исчисления высказыв
+
+\begin{lemma}
+Если $\alpha$ содержит $n$ пропозициональных переменных $P_1 \dots P_n$, то \\
+$_{[\llbracket P_1 \rrbracket]}P_1, \dots _{[\llbracket P_n \rrbracket]}P_n \vdash _{[\llbracket \alpha \rrbracket]}\alpha$
```
Вместо форсированного перевода тут нужно формулу писать внутри `\[ \]` вместо `$ $`.

artemohanjanyan commented on lecture3.tex:
```diff
@@ -319,6 +319,35 @@ \section{Теорема о полноте исчисления высказыв
+
+База: $\alpha = P$. Тогда очевидно, что $P \vdash P$, $\neg P \vdash \neg P$.
+
+Переход: Пусть утверждение уже доказано для любых формул, короче некоторой формулы $\alpha$. Докажем его для формулы $\alpha$.\\
```
`\\` нельзя

artemohanjanyan commented on lecture3.tex:
```diff
@@ -319,6 +319,35 @@ \section{Теорема о полноте исчисления высказыв
+Обозначим $_{[\llbracket P_1 \rrbracket]}P_1, \dots _{[\llbracket P_n \rrbracket]}P_n$ за $\Pi$.\\
+$\alpha \equiv \beta \star \gamma$. Тогда, так как формулы $\beta$ и $\gamma$ короче формулы $\alpha$, то к ним применимо предположение индукции, то есть $\Pi$ $\vdash$ $_{[\llbracket \beta \rrbracket]}\beta$ и $\Pi$ $\vdash$ $_{[\llbracket \gamma \rrbracket]}\gamma$.
+По лемме \ref{any_connection}, $_{[\llbracket \beta \rrbracket]}\beta, _{[\llbracket \gamma \rrbracket]}\gamma \vdash _{[\llbracket \alpha \rrbracket]}\alpha$.
+
```
Из-за этой пустой строки квадратик, обозначающий конец доказательства, стоит на отдельной пустой строчке.

turing228 committed:
Первый ответ

Убрал пустые строчки перед \end{proof}

Избавился от своих форсированных переводом

artemohanjanyan commented:
- Лемма 4.7 очень похожа на лемму 4.3.
- В теореме в переходе предполагается, что для некоего n-k>0 мы уже доказали, хотя в базе n-k=0.
- Очевидное упражнение я бы убрал.

artemohanjanyan commented on lecture3.tex:
```diff
@@ -319,6 +319,30 @@ \section{Теорема о полноте исчисления высказыв
+Если $\models \alpha$ и $\alpha$ содержит $n$ пропозициональных переменных $P_1 \dots P_n$, то $_{[x_1]} P_1, \dots _{[x_n]} P_n \vdash \alpha$ при любых $x_i$.
+\end{lemma}
+
+\begin{proof}Частный случай леммы \ref{derivability_with_propositionals} и определения $\models$.\end{proof}
```
Не на ту лемму ссылаешься

artemohanjanyan commented on lecture3.tex:
```diff
@@ -189,8 +189,8 @@ \section{Теорема о полноте исчисления высказыв
-\begin{lemma}
-Каждое из построенных по таблицам истинности утверждений доказуемо.
+\begin{lemma}\label{any_connection}
+$_{[\llbracket \alpha \rrbracket]} \alpha, _{[\llbracket \beta \rrbracket]} \beta \vdash _{[\llbracket \alpha \star \beta \rrbracket]} \alpha \star \beta$, где $\star$ --- любая связка.
```
```suggestion
$_{[\llbracket \alpha \rrbracket]} \alpha, _{[\llbracket \beta \rrbracket]} \beta \vdash _{[\llbracket \alpha \star \beta \rrbracket]} \alpha \star \beta$, где $\star$ "--- любая связка.
```

turing228 committed:
Второй коммит

Заменил 4.3 на 4.7 и убрал 4.7

В переходе n-k>=0 теперь (так как база для 0 доказывается)

Доказетельство очевидного утверждения 4.9 раскрыто.
